test/extensions/filters/http/ext_proc/ext_proc_integration_test.cc

```diff
@@ -95,12 +116,136 @@ class ExtProcIntegrationTest : public HttpIntegrationTest,
+      ASSERT_TRUE(processor_connection_->waitForNewStream(*dispatcher_, processor_stream_));
+    }
+    ASSERT_TRUE(processor_stream_->waitForGrpcMessage(*dispatcher_, request));
+    ASSERT_TRUE(request.has_response_headers());
+    if (first_message) {
+      processor_stream_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "200"}}, false);
+    }
+    ProcessingResponse response;
+    auto* headers = response.mutable_response_headers();
+    if (cb) {
+      (*cb)(request.response_headers(), *headers);
+    }
+    processor_stream_->sendGrpcMessage(response);
+  }
+
+  void processRequestBodyMessage(
+      bool first_message, absl::optional<std::function<void(const HttpBody&, BodyResponse&)>> cb) {
+    ProcessingRequest request;
+    if (first_message) {
+      ASSERT_TRUE(
+          fake_upstreams_.back()->waitForHttpConnection(*dispatcher_, processor_connection_));
+      ASSERT_TRUE(processor_connection_->waitForNewStream(*dispatcher_, processor_stream_));
+    }
+    ASSERT_TRUE(processor_stream_->waitForGrpcMessage(*dispatcher_, request));
+    ASSERT_TRUE(request.has_request_body());
+    if (first_message) {
+      processor_stream_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "200"}}, false);
+    }
+    ProcessingResponse response;
+    auto* body = response.mutable_request_body();
+    if (cb) {
+      (*cb)(request.request_body(), *body);
+    }
+    processor_stream_->sendGrpcMessage(response);
+  }
+
+  void processResponseBodyMessage(
+      bool first_message, absl::optional<std::function<void(const HttpBody&, BodyResponse&)>> cb) {
+    ProcessingRequest request;
+    if (first_message) {
+      ASSERT_TRUE(
+          fake_upstreams_.back()->waitForHttpConnection(*dispatcher_, processor_connection_));
+      ASSERT_TRUE(processor_connection_->waitForNewStream(*dispatcher_, processor_stream_));
+    }
+    ASSERT_TRUE(processor_stream_->waitForGrpcMessage(*dispatcher_, request));
+    ASSERT_TRUE(request.has_response_body());
+    if (first_message) {
+      processor_stream_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "200"}}, false);
+    }
+    ProcessingResponse response;
+    auto* body = response.mutable_response_body();
+    if (cb) {
+      (*cb)(request.response_body(), *body);
+    }
+    processor_stream_->sendGrpcMessage(response);
+  }
+
+  void processAndRespondImmediately(bool first_message,
+                                    absl::optional<std::function<void(ImmediateResponse&)>> cb) {
+    ProcessingRequest request;
+    if (first_message) {
+      ASSERT_TRUE(
+          fake_upstreams_.back()->waitForHttpConnection(*dispatcher_, processor_connection_));
+      ASSERT_TRUE(processor_connection_->waitForNewStream(*dispatcher_, processor_stream_));
+    }
+    ASSERT_TRUE(processor_stream_->waitForGrpcMessage(*dispatcher_, request));
+    if (first_message) {
+      processor_stream_->encodeHeaders(Http::TestResponseHeaderMapImpl{{":status", "200"}}, false);
```

Comment:
Can we use `startGrpcStream()` here and elsewheere?

Reply:
Sure! Thanks!